Lib/glyphsLib/classes.py

```diff
@@ -3227,46 +3283,54 @@ def exports(self):
     def exports(self, value):
         self.active = value
 
-    def _get_from_properties(self, key):
-        for p in self.properties:
-            if p.key == key:
-                return p.defaultValue
-        return None
-
     @property
     def familyName(self):
-        value = (
-            self._get_from_properties("familyNames")
+        return (
+            self.properties.get("familyNames")
             or self.customParameters["familyName"]
+            or self.parent.familyName
         )
-        if value:
-            return value
-        return self.parent.familyName
 
     @familyName.setter
     def familyName(self, value):
+        # TODO: Update this to write into either custom parameters or properties
+        # depending on the version.
         self.customParameters["familyName"] = value
 
     @property
     def preferredFamily(self):
-        value = self.customParameters["preferredFamily"]
-        if value:
-            return value
-        return self.parent.familyName
+        self.preferredFamilyName
 
     @preferredFamily.setter
     def preferredFamily(self, value):
+        self.preferredFamilyName = value
+
+    @property
+    def preferredFamilyName(self):
+        return (
+            self.properties.get("preferredFamilyNames")
+            or self.customParameters["preferredFamily"]
+            or self.parent.familyName
```

Comment:
@IvanUkhov I just tried to do the following from the Macro window in Glyphs.app for a font which doesn't define any localised "Typographic Family Names":

```python
print(Glyphs.font.instances[0].preferredFamily)
print(Glyphs.font.instances[0].preferredFamilyName)
```

I got:

```
FlippedSmartComponents
None
```

Which means that the `preferredFamilyName` getter should return None if there are no such properties; whereas the `preferredFamily` returs the default `familyName` even if the properites aren't set.

Reply:
Thank you for looking into this. I have restored `preferredFamily` and reduced `preferredFamilyName` to not add uncertain functionality unnecessarily.

Reply:
thanks, please see my other comments below